Project/Client/MessageUI.cpp

```diff
@@ -0,0 +1,210 @@
+    {
+        ImGui::TableSetupColumn("Log");
+        ImGui::TableHeadersRow();
+
+        for (int row = 0; row < m_vectLog.size(); row++)
+        {
+            //색상 설정
+            ImVec4 Color;
+   
+            bool SearchDisplay = false;
+            bool ColorDisplay = false;
+
+            CheckSearchDisplay(SearchDisplay, row);
+
+            CheckColorDispaly(ColorDisplay, row);
+
+            if (ColorDisplay && SearchDisplay)
+            {
+                string count = std::to_string(row);
+                string temp = m_vectLog[row].m_strMsg.c_str() + count;
+
+                ImGui::TableNextRow();
+                ImGui::TableSetColumnIndex(0);
+                ImGui::TextColored(SetLogColor(m_vectLog[row]), temp.c_str());
+
+             }
+
+            // 현재 스크롤 위치
+            float currentScrollY = ImGui::GetScrollY();
+            float maxScrollY = ImGui::GetScrollMaxY();
+
+            // 만약 사용자가 스크롤을 움직이지 않았고 스크롤이 최하단에 있었다면, 스크롤을 최하단으로 유지
+            if (m_fLastScrollY == maxScrollY && m_bResize)
+            {
+                ImGui::SetScrollHereY(1.0f);
+            }
+
+            // 스크롤 위치 업데이트
+            m_fLastScrollY = ImGui::GetScrollY();
+
+        }
+
+        ImGui::EndTable();
+    }
+
+    m_iLogCount = m_vectLog.size();
+    m_bResize = false;
+
+
+}
+
+ImVec4 MessageUI::SetLogColor(tLog Log)
+{
+    ImVec4 Color;
+
+    switch (Log.m_LogLv)
+    {
+    case Log_Level::INFO:
+        return Color = m_LogColor[(UINT)Log_Level::INFO];
+        break;
+    case Log_Level::WARN:
+        return Color = m_LogColor[(UINT)Log_Level::ERR];
+        break;
+    case Log_Level::ERR:
+        return Color = m_LogColor[(UINT)Log_Level::WARN];
+        break;
+    case Log_Level::END:
+        break;
+    default:
+        break;
+    }
+
+    return Color;
+}
+
+void MessageUI::UpdateData()
+{
+    // 데이터 갱신
+    m_vectLog = CLogMgr::GetInst()->GetLogvec();
+
+    // 사이즈 측정
+    m_Parent = dynamic_cast<LogUI*>(this->GetParentUI());
+
+    //필터값 갱신
+    m_charLogMsgFilter = m_Parent->GetLogTextFilter();
+
+
+    // 부모 ui 사이즈 받아오기
+    if (m_Parent != nullptr)
+        m_TableSize = m_Parent->GetUISize();
+}
+
+void MessageUI::CheckLogLvFilter()
+{
+    for (int i = 0; i < 4; ++i)
+    {
+        bool* mask = m_Parent->GetLogLvFilter();
+
+        if (mask[i] == true)
+        {
+            m_iLoglvMask |= (1 << i);
+        }
+        else
+        {
+            m_iLoglvMask &= ~(1 << i);
+        }
+    }
+}
+
+void MessageUI::CheckSearchDisplay(bool& check, int row)
+{
+    if (m_charLogMsgFilter.empty())
+    {
+        check = true;
+    }
+    else
+    {
+        for (int i = 0; i <= m_vectLog[row].m_strMsg.size() - m_charLogMsgFilter.size(); ++i)
+        {
+            if (m_vectLog[row].m_strMsg.substr(i, m_charLogMsgFilter.size()) == m_charLogMsgFilter)
+            {
+                check = true;
+                break;
+            }
+        }
+    }
+}
```

Comment:
필히 수정 필요 -> 파라미터, 리턴 변경으로 호출부도 변경 필요할 듯 싶고, 애초에 함수화가 필요 없어보임
```suggestion
bool MessageUI::CheckSearchDisplay(const string& str)
{
    return str.find(m_charLogMsgFilter) == string::npos ? false : true;
}
```